akka-actor-typed-tests/src/test/scala/akka/actor/typed/DeadLetterSpec.scala

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2017-2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed
+
+import akka.actor.IllegalActorStateException
+import akka.actor.testkit.typed.scaladsl.LogCapturing
+import akka.actor.testkit.typed.scaladsl.ScalaTestWithActorTestKit
+import akka.actor.typed.scaladsl.AskPattern.Askable
+import akka.actor.typed.scaladsl.AskPattern.schedulerFromActorSystem
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.scaladsl.Behaviors._
+import akka.util.Timeout
+import org.scalatest.wordspec.AnyWordSpecLike
+
+import java.util.concurrent.CountDownLatch
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.concurrent.TimeoutException
+import scala.concurrent.duration._
+import scala.util.Failure
+import scala.util.Success
+
+sealed trait Command
+case class Multiply(a: Int, b: Int, forwardRef: ActorRef[WorkerCommand], replyTo: ActorRef[Int]) extends Command
+case class ReplyResult(num: Int, replyTo: ActorRef[Int]) extends Command
+case class Ignore() extends Command
+
+sealed trait WorkerCommand
+case class WorkerMultiply(a: Int, b: Int, replyTo: ActorRef[WorkerResult]) extends WorkerCommand
+case class WorkerResult(num: Int) extends WorkerCommand
+
+class ManualTerminatedTestSetup(val workerLatch: CountDownLatch) {
```

Comment:
I don't think either test here requires blocking with a latch? Asking a terminated actor and ask without any reply leading to timeout should be possible without this kind of infrastructure.

Reply:
The original purpose of this was to simulate real-life scenarios. However, since you mentioned it, the testing can be greatly simplified by using probes to replace some interactions.

Reply:
@johanandren PTAL, I have pushed a fixes commit, to ensure the testing remains in minimal implementation.